libs/common/include/helpers/mathFuncs.h

```diff
@@ -70,7 +70,8 @@ constexpr T interpolate(const T startVal, const T endVal, const U elapsedTime, c
 }
 
 /// Linear interpolation, similar to C++20's std::lerp()
-constexpr float lerp(const float startVal, const float endVal, const float ratio) noexcept
+template<typename T>
+constexpr T lerp(const T startVal, const T endVal, const T ratio) noexcept
```

Comment:
Shall we use a 2nd template param for the ratio and maybe an enable-if to be a floating point type. That would allow `lerp(int, int, float)` and even `lerp(PointF, PointF, float)`. Might need making the return type `auto` as e.g.`int*float=float` and `Point*float = PointF`   
Might be confusing though, what do you think? At least allowing different float types for T and ratio looks reasonable and the PointF-float could be used here already.

Reply:
Not sure how to go about it in `lerp(int, int, float)` scenario - what would user expect in the return type? `float` as the value is somewhere between ints, or `int` as the input from/to were passed with that type? I can see scenarios where both ways would be preferred.

I'd leave it as simple as it is, and if given use case emerges then perhaps expand template thingy.